app/models/user.rb

```diff
@@ -218,21 +223,31 @@ def report_comment(report, content)
   end
   # endregion
 
-  # forwards fill
   def banned?
-    self.permanently_banned? or ((not self.banned_until.nil?) and self.banned_until >= DateTime.current)
+    self.bans.current.count > 0
   end
 
   def unban
-    self.update(permanently_banned: false, ban_reason: nil, banned_until: nil)
+    UseCase::User::Unban.call(id)
   end
 
-  def ban(buntil=nil, reason=nil)
-    if buntil == nil
-      self.update(permanently_banned: true, ban_reason: reason)
+  # Bans a user.
+  # @param duration [Fixnum, nil] Ban duration
```

Comment:
Fixnum doesn't really exist these days, `Integer` is where it's at
```suggestion
  # @param duration [Integer?] Ban duration
```

Reply:
669aa579